engine/timekeeper.go

```diff
@@ -20,50 +21,40 @@ var (
 
 // ntpManager starts the NTP manager
 type ntpManager struct {
-	started       int32
-	stopped       int32
-	inititalCheck bool
-	shutdown      chan struct{}
+	started      int32
+	initialCheck bool
+	shutdown     chan struct{}
 }
 
 func (n *ntpManager) Started() bool {
 	return atomic.LoadInt32(&n.started) == 1
 }
 
-func (n *ntpManager) Start() (err error) {
-	if atomic.AddInt32(&n.started, 1) != 1 {
-		return errors.New("NTP manager already started")
+func (n *ntpManager) Start() error {
+	if !atomic.CompareAndSwapInt32(&n.started, 0, 1) {
+		return fmt.Errorf("NTP manager %w", subsystem.ErrSubSystemAlreadyStarted)
 	}
 
-	var disable bool
-	defer func() {
-		if err != nil || disable {
-			atomic.CompareAndSwapInt32(&n.started, 1, 0)
-		}
-	}()
-
 	if Bot.Config.NTPClient.Level == -1 {
-		err = errors.New("NTP client disabled")
-		return
+		atomic.CompareAndSwapInt32(&n.started, 1, 0)
+		return errors.New("NTP client disabled")
 	}
 
 	log.Debugln(log.TimeMgr, "NTP manager starting...")
 	if Bot.Config.NTPClient.Level == 0 && *Bot.Config.Logging.Enabled {
 		// Initial NTP check (prompts user on how we should proceed)
-		n.inititalCheck = true
-
+		n.initialCheck = true
 		// Sometimes the NTP client can have transient issues due to UDP, try
 		// the default retry limits before giving up
 		for i := 0; i < NTPRetryLimit; i++ {
-			err = n.processTime()
+			err := n.processTime()
 			switch err {
 			case nil:
 				break
```

Comment:
This looks like it breaks the switch case might want to break the for loop?

Reply:
haha sweet pickup of some old codes